src/Microsoft.Extensions.Options.ConfigurationExtensions/ConfigureFromConfigurationOptions.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.Extensions.Options
 {
-    public class ConfigureFromConfigurationOptions<TOptions> : ConfigureOptions<TOptions>
+    public class ConfigureFromConfigurationOptions<TOptions> : ConfigureOptions<TOptions>, IOptionsConfiguration<TOptions>
```

Comment:
I like in principle the renaming of the interface very much. Using a verb in `IConfigureOptions` made it hard to name things that interacted with it or that implemented it. 

I wonder if this goes far enough though. The main remaining problem I believe is that we are overloading the word Configuration to mean `Microsoft.Extensions.Configuration` and also "the thing that you do to mutate Options". 

That makes it almost impossible to find a good name for this class. I.e. `ConfigurationOptionsConfiguration` would probably be the name we would come up with if we just applied the standard rules to name this class: it is just an `OptionsConfiguration` that is based on `Configuration`.

I wonder if we can come up with a better word for "the thing that you do to mutate options" and apply it to the interface name and the service collection extension, e.g.:
- `IOptionsSetup` the original name we had wasn't that bad IMO
- `IOptionsMutator` it is probably the most mathematically correct but feels too obscure.

Any thoughts on this?